### ISSUE TITLE
Runtime crash when the app instance is recreated after getting destroyed by the system

### ISSUE THREAD

**Issue by ZahidRasheed** [InsertKoinIO/koin#386]
**A scope with id 'MyAwesomeActivity' already exists. Reuse or close it.**

`Caused by org.koin.a.c.g: A scope with id 'MyAwesomeActivity' already exists. Reuse or close it.
       at org.koin.core.registry.ScopeRegistry.loadScopes$koin_core(SourceFile:71)
       at org.koin.core.registry.ScopeRegistry.loadScopes$koin_core(SourceFile:65)
       at org.koin.core.Koin.createScope(SourceFile:152)
       at org.koin.android.scope.LifecycleOwnerExtKt.createScope(SourceFile:55)
       at org.koin.android.scope.LifecycleOwnerExtKt.createScope$default(SourceFile:54)
       at com.company.MyAwesomeActivity.onCreate(SourceFile:62)
       at android.app.Activity.performCreate(Activity.java:6977)
       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1126)
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2946)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3064)
       at android.app.ActivityThread.-wrap14(ActivityThread.java)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1659)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:154)
       at android.app.ActivityThread.main(ActivityThread.java:6823)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1563)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1451)`

Unable to reproduce it even with "Don't keep activities" enabled in the Developer options. Stack trace is through crashlytics. 

**Expected behavior**
Should delete the scope when activity is destroyed. 

**Koin project used and used version (please complete the following information):**
`2.0.0-beta-1`

**Comment by arnaudgiuliani** [comment]
Hello,

do you use the `bindScope` function to bind your scope to your Activity's lifecycle?

**Comment by ZahidRasheed** [comment]
Inside onCreate I am creating scope like below;

`bindScope(createScope(BRANDED_FRONT_DOOR_SCOPE))`

**Comment by erickok** [comment]
Are you lookling for `bindScope(getActivityScope())` perhaps?

**Comment by ZahidRasheed** [comment]
Ok above make sense now. getActivityScope() actually get or create. Thanks for the feedback.

**Comment by omar-ejv** [comment]
there seems to be still a bug when creating an activity from two different processes in Android.
1. let's say we have the HomeLancher process that starts the 'MyAwesomeApp', which then starts 'MyAwesomeActivity'. My activity in this case is not single-instance, so you can have multiple instances of it running simultaneously. We will keep the activity and app in the foreground and we go to step 2 below.
2. then we have another application that can also start the my-awesome-app. e.g. a camera app that starts my-awesome-app from the camera process.
3. when MyAwesomeActivity reaches onCreate on the , if we call `getActivityScope()`as far as I can read in the `org/koin/android/scope/LifecycleOwnerExt.kt.kt` it will try to find if the scope exist, based on the instance of the lifecykle owner, which is the activity in this case (scope.id). However, since my activity is not a single-instance there will be multiple instances of the activity itself.
Here is where Koin crashes because koin will attempt to create a new scope, which already exist.
In other words, there is a mismatch between scope.name and scope.id. You can have two scopes with different ids (instances of lifecycleowner) but they share the same name.

is this a bug or there is another way to handle this scenario?

**Comment by omar-ejv** [comment]
Actually looking at this again, I see that the id is the actual name of the class (not the instance).
We can still see the crash in the scenario above, but then I do not know where the problem could be :)
